js/github.js

```diff
@@ -9,10 +14,36 @@ const doesGitHubCookieExist = () => {
   return !!githubCookieStr;
 };
 
+const signInGitHubOAuth = () => {
+  const originURL = window.location.origin;
+  if (!originURL) return GITHUB_OAUTH;
+  return GITHUB_OAUTH + '&state=' + originURL;
+};
+
 const updateGitHubLink = () => {
+  const githubMock = JSON.parse(localStorage.getItem(GITHUB_OBJECT_KEY)) ?? {};
+  const isGithubMockExpired =
+    new Date().getTime() > (githubMock.expiresIn ?? 0);
+
+  let signUpLink = GITHUB_MOCK_URL;
+  if (githubMock.visited && !isGithubMockExpired) {
+    signUpLink = signInGithubOAuth();
```

Comment:
Please update the function name on Line 30 where you have called it `signInGitHubOAuth` and verify if changes are working fine or not